packages/common/src/components/TypeaheadSelect/TypeaheadSelect.tsx

```diff
@@ -0,0 +1,425 @@
+      | React.MouseEvent<Element, MouseEvent>
+      | React.KeyboardEvent<HTMLInputElement>
+      | undefined,
+    selection: string | number,
+  ) => void;
+  /** Callback triggered when the select opens or closes. */
+  onToggle?: (nextIsOpen: boolean) => void;
+  /** Callback triggered when the text in the input field changes. */
+  onInputChange?: (newValue: string) => void;
+  /** Function to return items matching the current filter value */
+  filterFunction?: (
+    filterValue: string,
+    options: TypeaheadSelectOption[],
+  ) => TypeaheadSelectOption[];
+  /** Callback triggered when the clear button is selected */
+  onClearSelection?: () => void;
+  /** Flag to allow clear current selection */
+  allowClear?: boolean;
+  /** Placeholder text for the select input. */
+  placeholder?: string;
+  /** Flag to indicate if the typeahead select allows new items */
+  isCreatable?: boolean;
+  /** Flag to indicate if create option should be at top of typeahead */
+  isCreateOptionOnTop?: boolean;
+  /** Message to display to create a new option */
+  createOptionMessage?: string | ((newValue: string) => string);
+  /** Message to display when no options are available. */
+  noOptionsAvailableMessage?: string;
+  /** Message to display when no options match the filter. */
+  noOptionsFoundMessage?: string | ((filter: string) => string);
+  /** Flag indicating the select should be disabled. */
+  isDisabled?: boolean;
+  /** Width of the toggle. */
+  toggleWidth?: string;
+  /** Additional props passed to the toggle. */
+  toggleProps?: MenuToggleProps;
+}
+
+const defaultNoOptionsFoundMessage = (filter: string) => `No results found for "${filter}"`;
+const defaultCreateOptionMessage = (newValue: string) => `Create "${newValue}"`;
+const defaultFilterFunction = (filterValue: string, options: TypeaheadSelectOption[]) =>
+  options.filter((o) => String(o.content).toLowerCase().includes(filterValue.toLowerCase()));
+
+export const TypeaheadSelect: React.FC<TypeaheadSelectProps> = ({
+  innerRef,
+  selectOptions,
+  onSelect,
+  onToggle,
+  onInputChange,
+  filterFunction = defaultFilterFunction,
+  onClearSelection,
+  allowClear,
+  placeholder = 'Select an option',
+  noOptionsAvailableMessage = 'No options are available',
+  noOptionsFoundMessage = defaultNoOptionsFoundMessage,
+  isCreatable = false,
+  isCreateOptionOnTop = false,
+  createOptionMessage = defaultCreateOptionMessage,
+  isDisabled,
+  toggleWidth,
+  toggleProps,
+  ...props
+}: TypeaheadSelectProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [filterValue, setFilterValue] = React.useState<string>('');
+  const [isFiltering, setIsFiltering] = React.useState<boolean>(false);
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(null);
+  const [activeItemId, setActiveItemId] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+
+  const NO_RESULTS = 'no results';
```

Comment:
nit pick: this string will be hard to translate, e.g. a user of this component will not be able to replace it with translated string or do other customization